plugin/nix-exprs/sbtix.nix

```diff
@@ -110,23 +115,28 @@ in rec {
             linkArtifact = outputPath: urlAttrs:
                 let
                   artifact =
-                    (if builtins.substring 0 5 urlAttrs.url == "file:" then
+                    if urlAttrs.type or null == "built" then
                       if localBuildsRepo == "" then
                         abort "'sbtixBuildInputs' parameter missing from 'buildSbtProgram'/'buildSbtLibrary', but local dependencies found."
                       else
-                        # replace the nix store path prefix to make sure
-                        # a repo.nix generated with a different version of
-                        # nixpkgs but the same dependencies will still work:
-                        localBuildsRepo + "/" + (lib.strings.concatStringsSep "/" (lib.lists.drop 4 (lib.strings.splitString "/" urlAttrs.url)))
+                        # Unlike fetched JARs which are content addressed derivations by virtue of being fixed-output,
+                        # these copies are not content addressed, because ca-derivations is still experimental.
+                        # This is unfortunate, because it will create duplication in the store during builds and development,
+                        # but at least these can be GC-ed and allow the end result to only reference the single JARs that
+                        # result from this copying operation.
+                        copyFile (baseNameOf urlAttrs.path) (localBuildsRepo + "/" + urlAttrs.path)
                     else
-                      fetchurl urlAttrs
-                    );
+                      fetchurl urlAttrs;
+                  hashBash =
+                    if urlAttrs.type or null == "built"
+                    then ''$(sha256sum "${artifact}" | cut -c -64)''
+                    else ''$(echo ${toLower urlAttrs.sha256} | tr / _)'';
```

Comment:
@raboof We could use `sha256sum` as an alternative here too, until the hash conversion functions are standardized. Does have a performance cost though.

Refs https://github.com/natural-transformation/sbtix/pull/20#issuecomment-1751109114

Reply:
Possible hash format conversion primops
- https://github.com/NixOS/nix/pull/7708

Reply:
using the `urlAttrs.sha256` seems OK to me